### ISSUE TITLE
Database Media Storage : php bin/magento catalog:images:resize fails to process images from database if not present locally (Again!)

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#23911]
### Preconditions (*)
1. magento 2.3-develop vanilla

### Steps to reproduce (*)
1. Deploy 2.3-develop

```
Stores -> Configuration
    Advanced -> System
        Storage Configuration for Media
            Media Storage = Database
            Select Media Database = default_setup
            Synchronize
            Save Config
Catalog -> Products
    Add Product
        Name = Test
        Price = 111
        Images and Videos
            Upload image "a.jpg"
        Save & Close
```
Verify image exists in filesystem
```
www-data@dev:~/dev1$ find pub/media -name *.jpg
pub/media/catalog/product/a/_/a.jpg
pub/media/catalog/product/cache/44df7a05cd543e344cd804fe1b7d03f9/a/_/a.jpg
pub/media/catalog/product/cache/05700e45ea23c173c607a07e57b31f14/a/_/a.jpg
pub/media/catalog/product/cache/c8b969a5db7b942d7c1ade87824d9fb5/a/_/a.jpg
pub/media/catalog/product/cache/8e4e1244680d07f5e23aac7a094b5ced/a/_/a.jpg
pub/media/catalog/product/cache/a73dbc1e401544f4a37404df685e56cb/a/_/a.jpg
pub/media/catalog/product/cache/565aa9b4f47b76913eb467f409e405a5/a/_/a.jpg
pub/media/catalog/product/cache/cea075b4acfaff3655c89768cb020a48/a/_/a.jpg
pub/media/catalog/product/cache/43b1d9aa96ae2ac357312f5e6f063ebb/a/_/a.jpg
pub/media/catalog/product/cache/2ac5ce82894659a1de39122f4ce5d785/a/_/a.jpg
pub/media/catalog/product/cache/efabf5d6ed985597d80c0c750d8a1488/a/_/a.jpg
pub/media/catalog/product/cache/927837b6328eeac4b65eb4a0cfed0c8d/a/_/a.jpg
pub/media/catalog/product/cache/42f22ea9d6381753b704516955973a32/a/_/a.jpg
pub/media/catalog/product/cache/a51cda8ad4b127ab6c26bcd10cb133c0/a/_/a.jpg
pub/media/catalog/product/cache/7bd4e189723785d9abc492cb6290143f/a/_/a.jpg
pub/media/catalog/product/cache/f6642d3506661139d447769aff06b0d0/a/_/a.jpg
pub/media/catalog/product/cache/82b662b5f44528d0717ba6faf8db557a/a/_/a.jpg
pub/media/catalog/product/cache/2b8fe12c3ac8320dcc99b17d8210084a/a/_/a.jpg
pub/media/catalog/product/cache/962b6a57cf93a71105bd781a2d0b61fc/a/_/a.jpg
pub/media/catalog/product/cache/9527c85c0eb98f85df3f97110b56ccfc/a/_/a.jpg
pub/media/catalog/product/cache/4ee5eeb329f91d52118bd6ece2f76036/a/_/a.jpg
www-data@dev:~/dev1$
```
Verify image exists in database
```
mysql> select filename,directory from media_storage_file_storage;
+----------------------------------+------------------------------------------------------------+
| filename                         | directory                                                  |
+----------------------------------+------------------------------------------------------------+
| preview_image_5d3b1f777bdc2.jpeg | theme/preview                                              |
| preview_image_5d3b1f77876c4.jpeg | theme/preview                                              |
| a.jpg                            | catalog/product/a/_                                        |
| a.jpg                            | catalog/product/cache/9527c85c0eb98f85df3f97110b56ccfc/a/_ |
| a.jpg                            | catalog/product/cache/05700e45ea23c173c607a07e57b31f14/a/_ |
| a.jpg                            | catalog/product/cache/f6642d3506661139d447769aff06b0d0/a/_ |
| a.jpg                            | catalog/product/cache/8e4e1244680d07f5e23aac7a094b5ced/a/_ |
| a.jpg                            | catalog/product/cache/a51cda8ad4b127ab6c26bcd10cb133c0/a/_ |
| a.jpg                            | catalog/product/cache/42f22ea9d6381753b704516955973a32/a/_ |
| a.jpg                            | catalog/product/cache/7bd4e189723785d9abc492cb6290143f/a/_ |
| a.jpg                            | catalog/product/cache/927837b6328eeac4b65eb4a0cfed0c8d/a/_ |
| a.jpg                            | catalog/product/cache/44df7a05cd543e344cd804fe1b7d03f9/a/_ |
| a.jpg                            | catalog/product/cache/cea075b4acfaff3655c89768cb020a48/a/_ |
| a.jpg                            | catalog/product/cache/565aa9b4f47b76913eb467f409e405a5/a/_ |
| a.jpg                            | catalog/product/cache/82b662b5f44528d0717ba6faf8db557a/a/_ |
| a.jpg                            | catalog/product/cache/43b1d9aa96ae2ac357312f5e6f063ebb/a/_ |
| a.jpg                            | catalog/product/cache/2b8fe12c3ac8320dcc99b17d8210084a/a/_ |
| a.jpg                            | catalog/product/cache/c8b969a5db7b942d7c1ade87824d9fb5/a/_ |
| a.jpg                            | catalog/product/cache/efabf5d6ed985597d80c0c750d8a1488/a/_ |
| a.jpg                            | catalog/product/cache/4ee5eeb329f91d52118bd6ece2f76036/a/_ |
| a.jpg                            | catalog/product/cache/a73dbc1e401544f4a37404df685e56cb/a/_ |
| a.jpg                            | catalog/product/cache/2ac5ce82894659a1de39122f4ce5d785/a/_ |
| a.jpg                            | catalog/product/cache/962b6a57cf93a71105bd781a2d0b61fc/a/_ |
+----------------------------------+------------------------------------------------------------+
23 rows in set (0.00 sec)

mysql>
```
Delete pub/media/catalog folder
```
www-data@dev:~/dev1$ rm -rf pub/media/catalog/
www-data@dev:~/dev1$
```
run php/magento catalog:image:resize to generate cached images

### Expected result (*)
Cached images generated in database and on file system

### Actual result (*)
```
www-data@dev:~/dev1$ php bin/magento catalog:images:resize
1/1 [============================] 100% < 1 sec 44.2 MiB        | /a/_/a.jpg
Product images resized with errors:
Cannot resize image "/var/www/dev1/pub/media/catalog/product/a/_/a.jpg" - original image not found
www-data@dev:~/dev1$
```

**Comment by magento-engcom-team** [comment]
@gwharton Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [x] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess
- [x] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18708` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @gwharton, @gwharton. 

Thank you for your report and collaboration!

The issue was fixed by @gwharton in magento/magento2#23913

The fix will be available with the upcoming `2.3.4` release.
